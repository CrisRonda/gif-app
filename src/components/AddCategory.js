import React, { useEffect, useRef, useState } from "react";
import PropTypes from "prop-types";
import { Input, Select } from "react-rainbow-components";

const options = [
  { value: 10, label: "10" },
  { value: 20, label: "20" },
  { value: 30, label: "30" },
];

const AddCategory = ({ setParams }) => {
  const [inputValue, setInputValue] = useState("");
  const [limit, setLimit] = useState(10);
  const ref = useRef();
  useEffect(() => {
    if (ref.current) {
      ref.current.focus();
    }
  }, []);

  // useEffect(() => {
  //   if (inputValue) setParams({ category: inputValue, limit });
  // }, [inputValue, limit, setParams]);
  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();

    if (inputValue.trim().length >= 2) {
      setParams({ category: inputValue, limit });
      setInputValue("");
    }
  };

  return (
    <form
      onSubmit={handleSubmit}
      style={{ display: "flex", justifyContent: "center" }}
    >
      <Input
        id="input"
        ref={ref}
        label="Search"
        style={{ maxWidth: 700, width: "80%" }}
        placeholder="Search your Gift"
        bottomHelpText="ex: Shaman King"
        value={inputValue}
        onChange={handleInputChange}
      />
      <Select
        label="By"
        value={limit}
        onChange={({ target: { value } }) => setLimit(value)}
        options={options}
        style={{ maxWidth: 356, marginLeft: 8 }}
        // className="rainbow-m-vertical_x-large rainbow-p-horizontal_medium rainbow-m_auto"
      />
    </form>
  );
};

AddCategory.propTypes = {
  setParams: PropTypes.func.isRequired,
};

export default AddCategory;
