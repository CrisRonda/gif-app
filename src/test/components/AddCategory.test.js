import { shallow } from "enzyme";
import AddCategory from "../../components/AddCategory";

describe("Test on AddCategory", () => {
  const setCategory = jest.fn();
  let wrapper = shallow(<AddCategory setParams={setCategory} />);
  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setParams={setCategory} />);
  });

  test("should show componente", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("should change value on input", () => {
    const input = wrapper.find("#input");
    input.simulate("change", { target: { value: "holi" } });
  });

  test("not should post info", () => {
    wrapper.find("form").simulate("submit", { preventDefault() {} });
    expect(setCategory).not.toHaveBeenCalled();
  });

  test("should set input & clear", () => {
    const value = "dark";
    const input = wrapper.find("#input");
    input.simulate("change", { target: { value } });
    wrapper.find("form").simulate("submit", { preventDefault() {} });
    expect(setCategory).toHaveBeenCalled();
    input.simulate("change", { target: { value } });
    expect(setCategory).toHaveBeenCalledTimes(1);
    expect(setCategory).toHaveBeenCalledWith(expect.any(Object));
  });
});
