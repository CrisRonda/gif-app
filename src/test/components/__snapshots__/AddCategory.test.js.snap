// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on AddCategory should show componente 1`] = `
<form
  onSubmit={[Function]}
  style={
    Object {
      "display": "flex",
      "justifyContent": "center",
    }
  }
>
  <_class
    bottomHelpText="ex: Shaman King"
    id="input"
    label="Search"
    onChange={[Function]}
    placeholder="Search your Gift"
    style={
      Object {
        "maxWidth": 700,
        "width": "80%",
      }
    }
    value=""
  />
  <_class
    label="By"
    onChange={[Function]}
    options={
      Array [
        Object {
          "label": "10",
          "value": 10,
        },
        Object {
          "label": "20",
          "value": 20,
        },
        Object {
          "label": "30",
          "value": 30,
        },
      ]
    }
    style={
      Object {
        "marginLeft": 8,
        "maxWidth": 356,
      }
    }
    value={10}
  />
</form>
`;
